consumer/src/main/kotlin/streams/procedures/StreamsSinkProcedures.kt

```diff
@@ -27,25 +29,40 @@ class StreamsSinkProcedures {
         val properties = config?.mapValues { it.value.toString() } ?: emptyMap()
         val configuration = StreamsSinkProcedures.streamsEventSinkConfigMapper.convert(config = properties)
 
-        val consumer = StreamsSinkProcedures
-                .streamsEventConsumerFactory
-                .createStreamsEventConsumer(configuration, log!!)
-                .withTopics(setOf(topic))
+        val data = readData(topic, config ?: emptyMap(), configuration)
+
+        return data.map { StreamResult(mapOf("data" to it)) }.stream()
+    }
+
+    private fun readData(topic: String, procedureConfig: Map<String, Any>, consumerConfig: Map<String, String>): List<Any> {
+        val cfg = procedureConfig.mapValues { if (it.key != "partitions") it.value else mapOf(topic to it.value) }
+        val timeout = cfg.getOrDefault("timeout", 1000).toString().toLong()
+        val consumer = createConsumer(consumerConfig, topic)
         consumer.start()
-        val data = try {
-            consumer.read()
+        val data = mutableListOf<Any>()
```

Comment:
I guess we'll have to rewrite that in the future into something async.
is consumer.read blocking?
if not this loop can burn a lot of cpu